runelite-client/src/main/java/net/runelite/client/util/WorldMapView.java

```diff
@@ -142,30 +114,65 @@ private static BufferedImage getMapRegion(Point worldPoint)
 		}
 	}
 
-	public Point[] worldPointsToGraphicsPoint(WorldPoint[] worldPoints)
+	/**
+	 * Updates a few simple calculations used by various methods.
+	 * Functions that use these variables are responsible for calling
+	 * this function to ensure they are up to date.
+	 */
+	private void updateCalculations()
+	{
+		//Calculate region graphical size.
+		regionSizeScaled = (int) (REGION_GRAPHIC_SIZE * zoom);
+		//How far from top left of region this tile is
+		tileOffset = new Point(center.getX() % REGION_TILE_SIZE, center.getY() % REGION_TILE_SIZE);
+		//Graphical size of tileOffset
+		pixelsPerTile = regionSizeScaled / (float)REGION_TILE_SIZE;
+		graphicalOffset = new Point((int)(tileOffset.x * pixelsPerTile), (int)(tileOffset.y * pixelsPerTile));
+		graphicalCenter = new Point(graphicsWidth / 2, graphicsHeight / 2);
+	}
+
+
+	public Point[] worldPointsToGraphicsPoint(WorldPoint... worldPoints)
 	{
+		updateCalculations();
 		Point[] retPoints = new Point[worldPoints.length];
 		for (int i = 0; i < worldPoints.length; i++)
 		{
-			retPoints[i] = worldPointToGraphicsPoint(worldPoints[i]);
+			retPoints[i] = _worldPointToGraphicsPoint(worldPoints[i]);
 		}
 		return  retPoints;
 	}
 
 	/**
-	 *
+	 * Convert a WorldPoint to a graphical position on the currently displayed map.
+	 * This will return the correct Point even if it is not visible on the current map.
 	 * @param worldPoint A WorldPoint to calculate the graphical position for
 	 * @return the graphical coordinates of the specified point.
 	 */
 	public Point worldPointToGraphicsPoint(WorldPoint worldPoint)
+	{
+		updateCalculations();
+		return _worldPointToGraphicsPoint(worldPoint);
+	}
+
+	/**
+	 * Convert points without updatingCalculations.
+	 * This method is only to be called when the calculations are already known
+	 * to be correct
+	 * @param worldPoint A WorldPoint to calculate the graphical position for
+	 * @return the graphical coordinates of the specified point.
+	 */
+	private Point _worldPointToGraphicsPoint(WorldPoint worldPoint)
```

Comment:
I don't think this is standard naming in java, maybe change it to something like worldPointToGraphicsPointWithoutCalulations? Or preferably, something shorter